src/services/CoresService.Video.cs

```diff
@@ -58,13 +85,50 @@ public void AddDisplayModes(string identifier)
     {
         var info = this.ReadCoreJson(identifier);
         var video = this.ReadVideoJson(identifier);
-        List<DisplayMode> all = ALL_MODES.Select(id => new DisplayMode { id = id }).ToList();
+        List<DisplayMode> all = new List<DisplayMode>();
+
 
         if (info.metadata.platform_ids.Contains("gb"))
         {
             all.AddRange(GB_MODES.Select(id => new DisplayMode { id = id }));
         }
 
+        if (info.metadata.platform_ids.Contains("gbc"))
+        {
+            all.AddRange(GBC_MODES.Select(id => new DisplayMode { id = id }));
+        }
+
+        if (info.metadata.platform_ids.Contains("gba"))
+        {
+            all.AddRange(GBA_MODES.Select(id => new DisplayMode { id = id }));
+        }
+
+        if (info.metadata.platform_ids.Contains("gg"))
+        {
+            all.AddRange(GG_MODES.Select(id => new DisplayMode { id = id }));
+        }
+
+        if (info.metadata.platform_ids.Contains("lynx"))
+        {
+            all.AddRange(LYNX_MODES.Select(id => new DisplayMode { id = id }));
+        }
+
+        if (info.metadata.platform_ids.Contains("ngp") || info.metadata.platform_ids.Contains("jtngp"))

```

Comment:
i haven't sat down to look closer at the whole PR to see if anything can be done differently, but

i think you should be able to do something like this to search the array for a substring, instead of an exact match
```csharp
if (info.metadata.platform_ids.Any(s => s.Contains("ngp")))
```

Reply:
I'm thinking regex matching would be better so that `ngp` doesn't match on `ngpc`

Reply:
good call, you're right